File: Swift/004_Program/main.swift
import Foundation

var numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

print("Original array: \(numbers)")

// Filter
let evenNumbers = numbers.filter { $0 % 2 == 0 }
print("Even numbers: \(evenNumbers)")

// Map
let squared = numbers.map { $0 * $0 }
print("Squared: \(squared)")

// Reduce
let sum = numbers.reduce(0, +)
print("Sum: \(sum)")

// Contains
print("Contains 5: \(numbers.contains(5))")

// Sort
let sorted = numbers.sorted(by: >)
print("Sorted descending: \(sorted)")

File: Kotlin/004_Program/Main.kt
fun main() {
    val numbers = listOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)

    println("Original list: $numbers")

    // Filter
    val evenNumbers = numbers.filter { it % 2 == 0 }
    println("Even numbers: $evenNumbers")

    // Map
    val squared = numbers.map { it * it }
    println("Squared: $squared")

    // Reduce
    val sum = numbers.reduce { acc, n -> acc + n }
    println("Sum: $sum")

    // Any, All, None
    println("Has even: ${numbers.any { it % 2 == 0 }}")
    println("All positive: ${numbers.all { it > 0 }}")
    println("None negative: ${numbers.none { it < 0 }}")
}
